src/test/java/ir/sahab/kafkaconsumer/EmbeddedKafkaServer.java

```diff
@@ -72,21 +68,14 @@ public void start() throws IOException, InterruptedException {
     }
 
     public void createTopic(String topicName, Integer numPartitions) {
-        ZkClient zkClient = null;
-        ZkUtils zkUtils;
-        try {
-            // We have to pass the serializer class to ZkClient constructor Otherwise createTopic will return
-            // without error. The topic will exist in zookeeper and be returned when listing topics, but Kafka
-            // itself does not create the topic.
-            zkClient = new ZkClient(zkServer.getAddress(), 30000, 30000, ZKStringSerializer$.MODULE$);
-            zkUtils = new ZkUtils(zkClient, new ZkConnection(zkServer.getAddress()), JaasUtils.isZkSecurityEnabled());
-            logger.info("Executing create Topic: " + topicName + ", partitions: " + numPartitions
-                    + ", replication-factor: 1.");
-            AdminUtils.createTopic(zkUtils, topicName, numPartitions, 1, new Properties(), Disabled$.MODULE$);
-        } finally {
-            if (zkClient != null) {
-                zkClient.close();
-            }
+        logger.info("Executing create Topic: " + topicName + ", partitions: " + numPartitions
+                + ", replication-factor: 1.");
+        try (Admin kafkaAdmin = Admin.create(singletonMap(BOOTSTRAP_SERVERS_CONFIG, getBrokerAddress()))) {
+            NewTopic topic = new NewTopic(topicName, numPartitions, (short) 1);
+            CreateTopicsResult result = kafkaAdmin.createTopics(singleton(topic));
+            result.all().get();
+        } catch (InterruptedException | ExecutionException e) {
+            throw new IllegalStateException("Can't create topic", e);
```

Comment:
IllegalStateException -> AssertionError

IllegelStateException means you have called me when it is not supposed to.